acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,53 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'C93063 - pxp-module-puppet run expect puppet failure' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment(app_type)
+
+  step 'On master, create a new environment that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  fail('intended failure')
+}
+SITEPP
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents" do
+    target_identities = []
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'), "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal('failed', action_result['status'], "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: '#{action_result["status"]}'")
```

Comment:
good idea, but the fail message doesn't come back in the responses:
```
[1] pry(#<Beaker::TestCase>)> responses
=> {"pcp://a2uylov6sb4vc6m.delivery.puppetlabs.net/agent"=>
  {:envelope=>
    {:id=>"e236526e-04bb-4c03-b06f-f11bce4b8a88",
     :targets=>["pcp://shell/ruby-pcp-client-6207"],
     :message_type=>"http://puppetlabs.com/rpc_blocking_response",
     :sender=>"pcp://a2uylov6sb4vc6m.delivery.puppetlabs.net/agent",
     :expires=>"2016-04-28T21:31:34.192626Z"},
   :data=>
    {"transaction_id"=>"73b45fc8-d76c-4021-8feb-05a08d0eb62b",
     "results"=>
      {"kind"=>"apply",
       "time"=>"2016-04-28 14:31:30 -0700",
       "transaction_uuid"=>"8e7d50ef-4ab1-41e5-878b-97a4b8c81dbc",
       "environment"=>"run_puppet_result_failed_bt3j2knm",
       "status"=>"failed",
       "exitcode"=>1,
       "version"=>1,
       "error_type"=>"agent_exit_non_zero",
       "error"=>"Puppet agent exited with a non 0 exitcode"}}}}
```

we could inspect the agent logs, but this is actually very difficult to do correctly, particularly on windows agents.